pkgs/by-name/br/broadcast-box/ignore-env-file.patch

```diff
@@ -0,0 +1,32 @@
+diff --git a/main.go b/main.go
+index 336fa70..5b3d1a4 100644
+--- a/main.go
++++ b/main.go
+@@ -16,7 +16,6 @@ import (
+ 
+ 	"github.com/glimesh/broadcast-box/internal/networktest"
+ 	"github.com/glimesh/broadcast-box/internal/webrtc"
+-	"github.com/joho/godotenv"
+ )
+ 
+ const (
+@@ -166,19 +165,6 @@ func corsHandler(next func(w http.ResponseWriter, r *http.Request)) http.Handler
+ }
+ 
+ func main() {
+-	if os.Getenv("APP_ENV") == "production" {
+-		log.Println("Loading `" + envFileProd + "`")
+-
+-		if err := godotenv.Load(envFileProd); err != nil {
+-			log.Fatal(err)
+-		}
+-	} else {
+-		log.Println("Loading `" + envFileDev + "`")
+-
+-		if err := godotenv.Load(envFileDev); err != nil {
+-			log.Fatal(err)
+-		}
+-	}
```

Comment:
I agree with removing the error message since they would be very misleading in the NixOS module. I'm not sure if we should remove support for loading `.env` files completely without any notice though. People can still run the service binarty without the NixOS module and might be opting to use it with an `.env` file as it is documented upstream.

Maybe we should still load the production env file if `$APP_ENV` is set to `production`? And additionally don't try load any `.env` file at all if `$APP_ENV` is set to an empty string or `nixos` and set this in the NixOS module?

Reply:
I agree that being able to respect `$APP_ENV` and load `.env` if they exist would be a nicer solution. Rather than removing that entire block of code in the patch I'll just swap the `log.Fatal(err)` lines with `log.Println(err)` in the case where `APP_ENV == "production"`. This way, broadcast box will only exit due to no `.env` file when `APP_ENV` has not been defined. This should help inform users (running the package outside of the NixOS module) that environment variables need to be set whilst still allowing the program to run in a service without a `.env` file.

Only downside is that the service in the NixOS module will log a harmless `".env.production file does not exist"` error on start-up. I could fix this by patching a special case for `$APP_ENV == "nixos"` but I'm not sure it's worth it?

Reply:
On second thought that error message was bothering me so I've modified the patch to check for `$APP_ENV == "nixos"`. Let me know if you'd rather avoid the extra patching.

Reply:
Broadcast box have now made `.env.production` the default instead of `.env.development` when `$APP_ENV` is undefined. To keep things simple I will just add a special case for `$APP_ENV == "nixos"` that doesn't attempt to load any `.env` files. 

Running broadcast box standalone without `$APP_ENV` set to `nixos` will require a `.env.production` file in the cwd. Otherwise, broadcast box quits with a `".env.production: no such file"` error. This matches upstream documentation so I think it's acceptable behaviour.